### DISCUSSION TITLE
Question about metadsl execution model

### DISCUSSION THREAD

**Discussion by dcharbon** [Quansight-Labs/metadsl#136] [Ideas]

**Comment by VimalKumarS** [comment]
Hi, I cam across metadsl library to be good fit for one my project , i tried example provided over wiki, and blog post over
https://labs.quansight.org/blog/2019/05/metadsl-dsl-framework/ ,  but from document still couldn't figure out the execute part, how to bind the actual function to be executed to get result from expression. can you please share some document / or a notebook for the same.  thanks

**Comment by saulshanabrook** [comment]
Hey @VimalKumarS. I haven't been working on metadsl recently.

If I was to try again, I would focus on supporting native Python control flow and enabling interop with existing libraries out of the box. Metadsl is a rather opinionated library and my take after working on it for a while is that although it could help existing DSL libraries in Python interop better together and reduce their duplication, without some shiny use cases out of the box it is just too complex to attract much investment!

I am curious what your project is that you were looking to use it for?
